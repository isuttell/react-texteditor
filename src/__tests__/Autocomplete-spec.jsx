/**
 * Test Autocomplete component
 */

import React from 'react';
import TestUtils from 'react-dom/test-utils';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import Immutable from 'immutable';

describe('Autocomplete', () => {
  const Autocomplete = require('../Autocomplete').default;

  it('should render without error', () => {
    const suggestions = new Immutable.OrderedMap({
      'rick': 'Rick',
      'morty': 'Morty'
    });
    let element = React.createElement(Autocomplete, {
      suggestions
    });

    expect(() => TestUtils.renderIntoDocument(element)).not.toThrow();
  });

  it('should match snapshot when suggestions are passed', () => {
    const suggestions = new Immutable.OrderedMap({
      'rick': 'Rick',
      'morty': 'Morty'
    });
    const wrapper = mount(
      <Autocomplete
        suggestions={suggestions}
        className='real-classy-stuff'
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot when an empty list of suggestions are passed', () => {
    const suggestions = new Immutable.OrderedMap();
    const wrapper = mount(
      <Autocomplete
        suggestions={suggestions}
        className='real-classy-stuff'
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot when suggestions are not passed', () => {
    const wrapper = mount(
      <Autocomplete
        className='real-classy-stuff'
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
