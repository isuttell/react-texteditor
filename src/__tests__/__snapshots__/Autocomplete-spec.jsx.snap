// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete should match snapshot when an empty list of suggestions are passed 1`] = `
<Autocomplete
  className="real-classy-stuff"
  onClick={null}
  suggestions={Array []}
>
  <div
    className="real-classy-stuff"
  >
    <div>
      No suggestions available.
    </div>
  </div>
</Autocomplete>
`;

exports[`Autocomplete should match snapshot when suggestions are not passed 1`] = `
<Autocomplete
  className="real-classy-stuff"
  onClick={null}
  suggestions={null}
/>
`;

exports[`Autocomplete should match snapshot when suggestions are passed 1`] = `
<Autocomplete
  className="real-classy-stuff"
  onClick={null}
  suggestions={
    Array [
      Object {
        "label": "Rick",
        "value": "rick",
      },
      Object {
        "label": "Morty",
        "value": "morty",
      },
    ]
  }
>
  <div
    className="real-classy-stuff"
  >
    <div>
      <div
        className="undefined"
        key="0"
        onClick={[Function]}
        onMouseOver={[Function]}
      >
        Rick
      </div>
      <div
        className=""
        key="1"
        onClick={[Function]}
        onMouseOver={[Function]}
      >
        Morty
      </div>
    </div>
  </div>
</Autocomplete>
`;
